sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -127,6 +127,9 @@ class ParquetFileFormat
       conf.setEnum(ParquetOutputFormat.JOB_SUMMARY_LEVEL, JobSummaryLevel.NONE)
     }
 
+    // PARQUET-1746: Disables page-level CRC checksums by default.
+    conf.setBooleanIfUnset(ParquetOutputFormat.PAGE_WRITE_CHECKSUM_ENABLED, false)
```

Comment:
This looks dangerous. Also cc @bbraams

Reply:
@wangyum Any chance you could elaborate on this a bit more? Are we convinced that the issue you pointed out in https://github.com/apache/spark/pull/26804#discussion_r561044576 is actually a regression caused by parquet and not a problem with the test itself (e.g. caused by any non-trivial assumptions made w.r.t. the output files)? Considering the benefit of having checksums enabled by default (e.g. much improved visibility into hard to debug data corruption issues), I'd propose further investigation before disabling the feature entirely and having Spark diverge from the `parquet-mr` defaults.

Regarding the defaults, support for checksums was added back in [PARQUET-1580](https://github.com/apache/parquet-mr/pull/647). These changes were included and released with `parquet-mr` 1.11.0 (see [CHANGES](https://github.com/apache/parquet-mr/blob/apache-parquet-1.11.0/CHANGES.md#version-1110)), and writing out checksums has been enabled by default since the release, see `ParquetProperties.java` in:
* [master](https://github.com/apache/parquet-mr/blob/master/parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java#L61)
* [apache-parquet-1.11.0](https://github.com/apache/parquet-mr/blob/apache-parquet-1.11.0/parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java#L54)
* [apache-parquet-1.11.1](https://github.com/apache/parquet-mr/blob/apache-parquet-1.11.1/parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java#L54)

I also noticed that [PARQUET-1746](https://issues.apache.org/jira/browse/PARQUET-1746) was raised and [a PR](https://github.com/apache/parquet-mr/pull/857) was opened for it to set the default to `false`, but that the issue has already been marked as resolved and the PR closed without merging the changes.

Reply:
1. Disable it to fix this regression: https://github.com/apache/spark/pull/26804#pullrequestreview-572328921.
2. Writing out checksums has minimal performance impact.
3. Do we really need this feature? I haven't seen Spark SQL users request this feature. This change just disable it by default, users can still enable this feature.

Reply:
I see it's been addressed in https://github.com/apache/spark/pull/26804/commits/72c52b64958340835e5a54b24aa68f201f4c15be, thanks for the quick fix @wangyum! 👍